src/App.js

```diff
@@ -1,5 +1,37 @@
+import CarsModel from "./CarsModel.js";
+import { ERROR_MESSAGES, MESSAGES } from "./constants/message.js";
+import {
+  getInputAsync,
+  printCarsMove,
+  printWinners,
+} from "./interfaceUtils.js";
+import { isCarNamesValid, isTrialInputValid } from "./validation.js";
+
 class App {
-  async run() {}
+  async run() {
+    const carsUserInput = await getInputAsync(MESSAGES.INPUT_CAR_NAMES);
+    const carsModel = new CarsModel(carsUserInput);
+
+    const carNames = carsModel.getCarNames();
+    const { isCarValid, errCarMessage } = isCarNamesValid(carNames);
+    if (!isCarValid) throw new Error(ERROR_MESSAGES.PREFIX + errCarMessage);
```

Comment:
`isCarValid` 가 필요할까? 라는 생각이 들었어요!

isCarNamesValid 함수를 보면 유효성 검사에 통과하지 못했을 때 isCarValid, errCarMessage를 리턴하고, 통과한 경우에는 isCarValid만 리턴합니다.
여기서 isCarValid는 유효성을 통과했는지 를 boolean 타입으로 나타내고 있는데, 이를 유효성이 통과한 경우에는 errCarMessage에 null 값을 할당하여 errCarMessage만 반환해도 충분히 isCarValid 역할을 할 수 있다고 생각했어요!

위처럼 구현하게 되면 변수 1개를 덜 사용할 수 있고, 유효성이 통과 된 케이스랑 통과되지 않은 케이스 둘다 동일하게 errCarMessage만 리턴하게 되어 가독성 측면에서도 좋다고 생각합니다!

만약 errCarMessage만 사용한다면 아래처럼 코드가 변경되겠네요!
```suggestion
    const errCarMessage = isCarNamesValid(carNames);
    if (!errCarMessage ) throw new Error(ERROR_MESSAGES.PREFIX + errCarMessage);
```

Reply:
저도 그렇게 작성해보았는데, 그렇게 되면 `isCarNamesValid`라는 함수명과 동떨어지게 되더라구요..!
이 함수는 에러메시지를 반환하기도 하지만 어쨌든 주된 목적은 **유효성 검사**니까요.

물론 축약도 되고 코드 줄이 줄어들 수는 있겠지만, 저는 의도를 더욱 명확하게 전달하기 위해서 이렇게 작성하게 되었습니다 ㅎㅎ
좋은 의견 감사드려요! 덕분에 저도 한번 더 생각해볼 수 있었습니다 👍